routing/router.go

```diff
@@ -2495,88 +2494,91 @@ func (r *ChannelRouter) sendToRoute(htlcHash lntypes.Hash, rt *route.Route,
 	// shard we'll now launch.
 	shardTracker := shards.NewSimpleShardTracker(htlcHash, nil)
 
-	// Launch a shard along the given route.
-	sh := &shardHandler{
-		router:       r,
-		identifier:   paymentIdentifier,
-		shardTracker: shardTracker,
-	}
-
-	var shardError error
-	attempt, outcome, err := sh.launchShard(rt, false)
-
-	// With SendToRoute, it can happen that the route exceeds protocol
-	// constraints. Mark the payment as failed with an internal error.
-	if err == route.ErrMaxRouteHopsExceeded ||
-		err == sphinx.ErrMaxRoutingInfoSizeExceeded {
-
-		log.Debugf("Invalid route provided for payment %x: %v",
-			paymentIdentifier, err)
+	// Create a payment lifecycle using the given route with,
+	// - zero fee limit as we are not requesting routes.
+	// - nil payment session (since we already have a route).
+	// - no payment timeout.
+	// - no current block height.
+	p := newPaymentLifecycle(
```

Comment:
I think they are not related? Why would you skip temp error for a non-MPP payment? To retry it if it fails the first time?

Reply:
Ok that sounds reasonable, but that we should restrict using the --skip_temp_err so that it can only be used when we attacked the MPP/AMP field ? TBH when I first used it I just tested with `queryroutes` and piped the result into `sendtoroutes`. After the payment failed, I was not able to remove this payment again. Only when registering a new Payment (which sets the MPP records). So we should check if the MPP record is present to make it more strict so that users cannot end up in this case or ?

Reply:
> After the payment failed, I was not able to remove this payment again.

You mean you sent a payment which was non-MPP but used the flag `skip_temp_err` right? What if you send it again without the flag, then I think the payment will be marked as failed?

I think they are not related in the sense that they are independent variables - you can send a non-MPP payment with this flag, which means you can retry the payment using a different route. Or you can send without the flag so this will be a one time thing. Also there's the `deletepayments` RPC you can use to remove it?